stonesoup/predictor/kalman.py

```diff
@@ -376,3 +393,78 @@ def predict(self, prior, timestamp=None, **kwargs):
+    output. The alternative, accessible via the :attr:`qr_method = True` flag, is to predict via a
+    modified Gram-Schmidt process. See [1] for details.
+
+    If transition and control models are possessed of the square root form of the covariance (as
+    :attr:`sqrt_covar` in the case of the transition model and :attr:`sqrt_control_noi` for
+    control models), then these are used directly. If not then they are created from the full
+    matrices using the scipy.linalg :meth:`sqrtm()` method. (Unlike the Cholesky decomposition
+    this works on positive semi-definite matrices, as well as positive definite ones.
+
+    References
+    ----------
+    1. Maybeck, P.S. 1994, Stochastic Models, Estimation, and Control, Vol. 1, NavtechGPS,
+       Springfield, VA.
+
+    """
+    qr_method = Property(bool, default=False, doc="A switch to do the prediction via a QR "
+                                                  "decomposition, rather than using a Cholesky"
+                                                  "decomposition.")
+
+    _prediction_class = SqrtGaussianStatePrediction
+
+    def _predicted_covariance(self, prior, predict_over_interval, **kwargs):
+        """Private function to return the predicted covariance.
+
+        Parameters
+        ----------
+        prior : :class:`~.SqrtGaussianState`
+            The prior class (which carries the covariance in square root form)
+        predict_over_interval : :class`~.timedelta`
+            The interval over which the model is applied
+
+        Returns
+        -------
+        : :class:`~.CovarianceMatrix`
+            The predicted covariance matrix
+
+        """
+        sqrt_prior_cov = prior.sqrt_covar
+
+        trans_m = self._transition_matrix(prior=prior, time_interval=predict_over_interval,
+                                          **kwargs)
+        if hasattr(self.transition_model, 'sqrt_covar'):
+            sqrt_trans_cov = self.transition_model.sqrt_covar(time_interval=predict_over_interval,
+                                                              **kwargs)
+        else:
+            sqrt_trans_cov = la.sqrtm(self.transition_model.covar(
+                time_interval=predict_over_interval, **kwargs))
+
+        # As this is Kalman-like, the control model must be capable of returning a control matrix
+        # (B)
+        ctrl_mat = self._control_matrix
+        if hasattr(self.control_model, 'sqrt_control_noise'):
+            sqrt_ctrl_noi = self.control_model.sqrt_control_noise
+        else:
+            sqrt_ctrl_noi = la.sqrtm(self.control_model.control_noise)
+
+        if self.qr_method:
+            # Note that the control matrix aspect of this hasn't been tested
+            m_sq_trans_cov = np.block([[trans_m @ sqrt_prior_cov, sqrt_trans_cov,
+                                        ctrl_mat@sqrt_ctrl_noi]])
+            [_, pred_sqrt_cov] = np.linalg.qr(m_sq_trans_cov.T)
```

Comment:
Brackets shouldn't be needed here
```suggestion
            _, pred_sqrt_cov = np.linalg.qr(m_sq_trans_cov.T)
```